src/code/compareSimulations.R

```diff
@@ -0,0 +1,498 @@
+        }
+        if(objectType == "Observer"){
+          formulaStringFirst = getObserverFormula(path_id = idFirst, DCI_Info = DCI_Info1)$Value;
+        }
+      }
+      
+      #Compare the objects
+      identical = (isTRUE(all.equal.numeric(firstValue, secondValue, tolerance = numTolerance)));
+      if (!ignoreFormula){
+        identical = (identical & (isFormulaInFirst == isFormulaInSecond) & isTRUE(all.equal.character(formulaStringFirst, formulaStringSecond)));
+      }
+    }
+    #Object is not in first simulation
+    else{
+      identical = FALSE;
+      firstValue = NaN;
+      isFormulaInFirst = NaN;
+      formulaStringFirst = "";
+    }
+    
+    #If the objects are not identical, create an entry in differences-list.
+    if( !identical ){
+      objectsPaths = c(objectsPaths, objectPath);
+      existanceInFirst = c(existanceInFirst, existsInFirst);
+      existanceInSecond = c(existanceInSecond, existsInSecond);
+      valuesFirst = c(valuesFirst, firstValue);
+      valuesSecond = c(valuesSecond, secondValue);
+      formulaFlagsFirst = c(formulaFlagsFirst, isFormulaInFirst);
+      formulaFlagsSecond = c(formulaFlagsSecond, isFormulaInSecond);
+      formulaStringsFirst = c(formulaStringsFirst, formulaStringFirst);
+      formulaStringsSecond = c(formulaStringsSecond, formulaStringSecond);
+    }
+  }
+  
+  objects_diff$Path = objectsPaths;
+  objects_diff$isExistingFirst = existanceInFirst;
+  objects_diff$isExistingSecond = existanceInSecond;
+  objects_diff$ValueFirst = valuesFirst;
+  objects_diff$ValueSecond = valuesSecond;
+  objects_diff$IsFormulaFirst = formulaFlagsFirst;
+  objects_diff$IsFormulaSecond = formulaFlagsSecond;
+  objects_diff$FormulaFirst = formulaStringsFirst;
+  objects_diff$FormulaSecond = formulaStringsSecond;
+  return(objects_diff);
+}
+
+compareSimulationTime = function(simTime1, simTime2){
+  identical = TRUE;
+  #Compare time pattern by size and element-wise
+  if ( !(length(simTime1$Time) == (length(simTime2$Time))) | !isTRUE(all.equal.numeric(simTime1$Time, simTime2$Time)) ){
+    identical = FALSE;
+  }
+  return(identical);
+}
+
+writeToCSV = function(CSV_path, results){
+  dirname <- dirname(CSV_path)
+  if (!file.exists(dirname)) {
+    dir.create(dirname, recursive=TRUE)
+  }
+  outString = "Simulation names";
+  outString = c(outString, paste(results$SimNames[1], results$SimNames[2], sep = "\t"));
+  
+  outString = c(outString, "Parameter differences:");
+  outString = c(outString, paste("Path", "Exists in first", "Exists in second", "Value in first", "Value in second", "Is formula in first", "Is formula in second", "Formula string in first", "Formula string in second", sep = "\t"));
+  if (length(results$Params_diff) > 0){
+    for (i in 1 : length(results$Params_diff$Path)){
+      outString = c(outString, paste(
+        results$Params_diff$Path[i],
```

Comment:
Would it be possible to create some functions to avoid code repetition? There are a lot of instances where the code ALMOST looks the same or is actually the same such as
```
  results$Species_diff$Path[i],
        results$Species_diff$isExistingFirst[i],
        results$Species_diff$isExistingSecond[i],
        results$Species_diff$ValueFirst[i],
        results$Species_diff$ValueSecond[i],
        results$Species_diff$IsFormulaFirst[i],
        results$Species_diff$IsFormulaSecond[i],
        results$Species_diff$FormulaFirst[i],
        results$Species_diff$FormulaSecond[i],
```
using Species_diff instead of Params_diff etc..